### ISSUE TITLE
Unable to sync. (Again)

### ISSUE THREAD

**Issue by lucia9096** [kakaroto/SWProxy#83]
So is it just me or we can't sync the swproxy with the game again prior to the latest application?

**Comment by daddyrubin** [comment]
@knifesk said  > If you double tap the home screen as soon as the "Connecting to game server" message appears and then switch to settings and set the proxy up, everything works as expected

This lets the proxy run for me but it does not create the files.

**Comment by lucia9096** [comment]
So it does connects but the files are not created. Anyone else has the same issue?

**Comment by azrethos** [comment]
yes seems ios is broken

**Comment by lucia9096** [comment]
How about android?

**Comment by Xzandro** [comment]
Android works fine.

**Comment by daddyrubin** [comment]
i guess i'll have to charge my Tab3 to get the files until it is updated :(

**Comment by knifesk** [comment]
@daddyrubin then try a moment early.. It works for me... I have to do it several times to get it to work.. the problem is that you're setting the proxy a little late, as the monsters and rune request is already finished

**Comment by daddyrubin** [comment]
@knifesk ok i will give if a few more tries.  Thanks.

**Comment by davidwang2000** [comment]
This used to work for me but it is broken since the last update.

**Comment by goku-ssj** [comment]
Does not work for me either since the new game patch. any news?

**Comment by CnoteLA** [comment]
Yeah, Im not able to get the files generated.  It starts up normally and then just hangs endlessly.

**Comment by Bithnar** [comment]
Same here. It gets to the Touch To Start then just hangs

**Comment by kupoback** [comment]
Same here does not work. Any other methods to try and get this to upload, besides manually? I really need some help on my account, and don't want to upload everything manually.

**Comment by lstern** [comment]
everybody using ios or any android user having this issue ?

**Comment by kupoback** [comment]
iOS 9.3.2

**Comment by rolares** [comment]
iOS 8.1.2; same issue of having the proxy work just fine, logs activity as normal - all except for generating optimizer.json and other files.

**Comment by skroll** [comment]
I was able to get it using daddyrubin's advice. If it doesn't generate the rune information, you gotta be a little bit quicker. It sucks, but it's the only way at the moment for it to work.

Looks like they're detecting a proxy somehow.

**Comment by lstern** [comment]
Sadly I don't have an IOS to test this.

**Comment by skroll** [comment]
I have a feeling that SW is using the CFProxySupport API to check if a proxy is set.

https://developer.apple.com/library/ios/documentation/CoreFoundation/Reference/CFProxySupport/index.html

Since iOS provides a way to check, they're probably actively trying to prevent it.

**Comment by kupoback** [comment]
No combination of turning proxy on when "Getting Friends List" or "Connecting to Game Server" as it gets to "Touch to Start" and hangs up trying to connect to the game. I've even tried 0.97 to see if that'll do anything either as that was the last working version for me.

**Comment by lstern** [comment]
@skroll given that com2us recently changed the crypto key a new update with proxy detection is likely. Could you test this configuring/using  a regular proxy and checking if game is ok ?

**Comment by knifesk** [comment]
@lstern I can confirm that with a fresh Squid install on a Centos 6.5 server that acts as my home gateway its working... Maybe the proxy modifies the HTTP headers and they are able to detect this specific proxy?

![image](https://cloud.githubusercontent.com/assets/333676/14798263/15d0c8c4-0b0d-11e6-89d0-a48f58dff824.png)

**Comment by c2uwillbanme** [comment]
@lstern I can confirm that with a fresh Squid install on a Centos 6.5 server that acts as my home gateway its working... Maybe the proxy modifies the HTTP headers and they are able to detect this specific proxy?

![image](https://cloud.githubusercontent.com/assets/18667718/14798607/dcca3aa4-0b0e-11e6-8a76-d2a836e920ce.png)

edit: Just checking the Log... I think this is the source of the problems...

`1461616849.885 32123 10.0.0.149 TCP_MISS/200 5857 CONNECT 169452.engine.mobileapptracking.com:443 - DIRECT/54.164.196.101 -`

an http verb that this proxy is probably not interpreting

In my squid log I can see this CONNECT and then some GET/POST calls... 
In my swproxy log I can see this CONNECT and then the log ends...

So, I think the problem may be that this proxy doesnt support CONNECT

**Comment by skroll** [comment]
@lstern Just used another proxy, and it worked just fine. I was able to sniff out the requests/responses, which I fed through the decryption routine and was able to get the raw JSON just fine. Didn't have to jump through hoops to get it to work.

**Comment by kupoback** [comment]
@skroll What do you mean by used another proxy? As opposed to 8080?

**Comment by skroll** [comment]
@lstern No, I mean I used an entirely different proxy tool besides SWProxy.

**Comment by goku-ssj** [comment]
and this tool is...?

**Comment by skroll** [comment]
Ah sorry, it was Charles Proxy: https://www.charlesproxy.com/

**Comment by lstern** [comment]
@skroll seems like a bug in the SW Proxy code, the code kakaroto got from https://github.com/abhinavsingh/proxy.py maybe you could isolate and send me the faulty request so I can replay it here and try to fix the bug

**Comment by skroll** [comment]
@lstern Sure, where should I send it?

**Comment by lstern** [comment]
http://expirebox.com/

**Comment by skroll** [comment]
Oh, putting a file up is no problem, I just don't want to send all my SW requests/responses (i'm assuming some login info is sent) to the world. Have an address I can send it to?

**Comment by lstern** [comment]
oh, send to my email leonardo.stern@gmail.com

**Comment by geeyummy** [comment]
I tried everything but its not working with ios.

**Comment by TuiKiken** [comment]
Maybe this will help you:
```
2016-04-28 20:48:13,624: proxy - INFO - 192.168.1.3:57472 - GET summonerswar-eu.com2us.net:80/api/location_c2.php - 200 OK - 331 bytes
2016-04-28 20:48:14,265: proxy - ERROR - Exception while receiving from connection client <socket._socketobject object at 0x1096e67c0> with reason error(54, 'Connection reset by peer')
Traceback (most recent call last):
  File "/Users/TuiKiken/Downloads/SWProxy-0.99.1/proxy.py", line 266, in recv
    data = self.conn.recv(bytes)
error: [Errno 54] Connection reset by peer
```

**Comment by lstern** [comment]
@TuiKiken no, this is a common error related to connection drops.

**Comment by lstern** [comment]
Any windows user with an ios device willing to help with the debug / fix process by running custom sw proxy versions and reporting (by line) if its working or not? add me LINE ID: fperegrinus

**Comment by lstern** [comment]
@c2uwillbanme, seems like you are right

![image](https://cloud.githubusercontent.com/assets/1051372/14920702/32ce262e-0e05-11e6-9438-e2cb6119f6e8.png)

**Comment by Nacheen81** [comment]
@lstern 

Same thing on my system for iOS. 

_____________________________
2016-04-30 16:26:46,650: SWPlugin - INFO - Loaded 5 plugins
2016-04-30 16:26:46,652: proxy - INFO - Starting server on 10.0.1.45, port 8080
2016-04-30 16:27:00,057: proxy - INFO - xxx.xxx.xxx.xxx:40716 - GET app-adforce.jp:80/ad//p/tmck?_app=2504&_bundle_id=com.com2us.smon.normal.freefull.apple.kr.ios.universal&_bv=2.1.1&_model=iPhone7%2C2&_os_ver=9.2.1&_sdk_ver=v2.15.2g - 200 OK - 488 bytes
2016-04-30 16:27:04,672: proxy - INFO - xxx.xxx.xxx.xxx:39410 - POST activeuser.qpyou.cn:80/gateway.php - 200 OK - 527 bytes
2016-04-30 16:27:04,689: proxy - INFO - xxx.xxx.xxx.xxx:38329 - POST push.qpyou.cn:80/api/agree/v1/get.php - 200 OK - 330 bytes
2016-04-30 16:27:05,236: proxy - INFO - xxx.xxx.xxx.xxx:33685 - POST push.qpyou.cn:80/api/register.php - 200 OK - 283 bytes
2016-04-30 16:27:05,239: proxy - INFO - xxx.xxx.xxx.xxx:38120 - POST activeuser.qpyou.cn:80/gateway.php - 200 OK - 463 bytes
2016-04-30 16:27:05,269: proxy - INFO - xxx.xxx.xxx.xxx:48319 - POST push.qpyou.cn:80/api/register.php - 200 OK - 283 bytes
2016-04-30 16:27:07,090: proxy - INFO - xxx.xxx.xxx.xxx:48943 - GET summonerswar-gb.com2us.net:80/api/location_c2.php - 200 OK - 331 bytes
2016-04-30 16:27:07,534: proxy - INFO - xxx.xxx.xxx.xxx:37100 - POST summonerswar-gb.com2us.net:80/api/server_status_c2.php - 200 OK - 287 bytes
2016-04-30 16:27:08,031: proxy - INFO - xxx.xxx.xxx.xxx:43838 - POST summonerswar-gb.com2us.net:80/api/version_info_c2.php - 200 OK - 1613 bytes
2016-04-30 16:27:08,311: proxy - INFO - xxx.xxx.xxx.xxx:33423 - GET image-glb.qpyou.cn:80/smon/hub_dist/res/download_2.00.86.inf - 200 OK - 983 bytes
2016-04-30 16:27:10,034: proxy - INFO - xxx.xxx.xxx.xxx:33696 - GET image-glb.qpyou.cn:80/smon/hub_dist/banapplist/banapplist.dat - 200 OK - 1524 bytes
2016-04-30 16:27:12,222: proxy - INFO - xxx.xxx.xxx.xxx:41539 - POST None:None/device/registered - None None - 0 bytes
2016-04-30 16:28:29,670: proxy - INFO - Closing server socket
___________________________


I was able to connect and successfully export the data by following the directions provided by @daddyrubin 

If I can provide additional logs or test options, I'm happy to assist.

**Comment by duduweizmann** [comment]
Sorry to get here without helping in any way. But do we have any outlook on when the problem will be solved? Saturday we have rune removal and I would love to have this working... 
Thanks...

**Comment by daddyrubin** [comment]
If you follow my instructions above you can get it to work.  Just double tap the home button and turn on the proxy right before it shows connecting to server

**Comment by kupoback** [comment]
@daddyrubin I've tried this many times but it simply does not work at all.

**Comment by duduweizmann** [comment]
@daddyrubin same for me. It simply does not work.

**Comment by Nacheen81** [comment]
It took several Artemis for me. But in got the timing right and it can work... I just successfully synced my data this morning to make sure it still worked.

**Comment by shorrockin** [comment]
Was in the same boat. The work around refused to work, until it did. So there is a timing you can get to generate the styles, but it's tricky.

**Comment by mockfake** [comment]
I tried this a number of times and couldn't get it to work, but tried the older PCAP approach and uploaded that file to Swarfarm -- also to no avail.  Finally gave up and logged in with an Android device that I had been using for an alt account, it's working like a charm.  The proxy would "work" fine for my iPhone, but I simply couldn't get the timing down to have it generate the rune data.

**Comment by kupoback** [comment]
Ok, so let me get this clear.

You start the game, and as soon as it says "Connecting to game server", you then double click home screen to bring up the Alt - Tab of iOS and turn the proxy on in settings. Am I understanding this right? Cuz, 70 times of trying this, I keep getting "Network connection error" over and over and over again.

**Comment by gzhou01** [comment]
Thanks! Got it to work with this. The key is to double tab the home right before it shows "Connecting to game server...". The timing to do this should be right after "Downloading necessary information...". As daddyrubin mentioned, it should be blank for a second before it shows "Connecting to game server..."

**Comment by thatrandomguyo1** [comment]
So to be clear this is still an ongoing issue a month later with no actual solution in sight other than hoping to hit some .1ms timing to change proxy over mid load?

**Comment by Nacheen81** [comment]
Yes. That's correct. 

Remember that you are welcome to join in and help find a solution to the problem and suggest an update to the code base. If you're unable due to a lack of knowledge about how to program a tool like this, show some patience and respect. Comments like yours,  @thatrandomguyo1, don't help to encourage those who have developed it to continue. 

Sorry for the rant, but if you're not here to help in finding a solution, keep your mouth shut.

**Comment by kakaroto** [comment]
@thatrandomguyo1 
Yes, to be clear, this is still an ongoing issue a month later in an open source program for which you could contribute the fix yourself if you wanted to, instead of asking for it on a silver platter. And yes, to be clear this is still an ongoing issue that only happens on iDevices which none of the developers seem to have, and yes, to be clear, this is still an ongoing issue in a program that gives something to the community without asking for anything in return, and which took time out of the developer's free time to develop and for which you're not supposed to expect any obligations from the developer's part.
Thank you for your feedback.

**Comment by skroll** [comment]
I have a solution to this but I don't know the best way to go about it. It's due to the proxy not supporting all methods, so there's a few solutions:

Fix the proxy lib in SWProxy.
  Pros: it's the lightest weight.
  Cons: not sure the code base lends itself to such modifications. I'm not much of a python guy (my professional background is C on embedded systems, Java on servers) to really go about this.

Use a more full fledged proxy lib.
  Pros: it's probably the easiest.
  Cons: it adds another dependency to the project, seems overkill for such a simple tool.

I've already been able to extract the rune data using a different proxy, then manually decrypting the results. It's definitely the proxy library here at work.

Any suggestions @

**Comment by skroll** [comment]
Btw the reason the CONNECT request is made is because of a https request. This is the mechanism for which regular http proxies hit https end points.

**Comment by thatrandomguyo1** [comment]
@Nacheen81 @kakaroto This issue is over 50 responses with varying responses half contradicting each other so I was looking for clarification since there's no actual response for it atm. So forgive me for asking a question in an open source forum where I don't have knowledge to contribute in any way. But I would be damn sure not to contribute if asking a question to an open ended problem a month later gets me nothing but contempt and "keep your mouth shut". Thanks for the work you've done in the past and forgive me for inquiring anything about the future of it.

**Comment by kupoback** [comment]
@thatrandomguyo1 Yes, the issue is over 50 responses long, however as @kakaroto stated, the devs don't have the tools to debug. You must be patient in the matters of an open source, free to use at your own risk and benefits, for a resolution.

Stating that if you had the knowledge and would contribute, but wouldn't due to your unneeded comment and upset towards the community may be a godsend to these developers.

Do not insult people who are here to help the community, of their own time and will. They are not making a dime off this and CANNOT be expected to dedicate all their free time to a resolution. Especially if they don't have the tools.

**Comment by skroll** [comment]
As a proof of concept, I spent 5 minutes banging this out in go:

https://gist.github.com/skroll/f4f46831daffb689de57b309ce36bf5d

Anyone else running iOS want to test this? It opens port 8080 (listening on all hosts) on your machine and works as a proxy, and just dumps "SW call" to stdout if it is proxying a call to the SW servers.

**Comment by Nacheen81** [comment]
@skroll  I'll load this on my server now and test it out.

**Comment by thatrandomguyo1** [comment]
I'm absolutely fine with being patient, I am not EXPECTING anything, I was simply asking if this is the current state of things. A month is a long time for any developer to not have any more follow up so it's a surprising state of affairs to not see any official change. If all the devs don't have said hardware that's 100% fine, I guess I missed the hardware listing on the way in. FRR day came up and this is a valuable tool so I asked as the current state of the tool is more or less unusable for iOS user. If that's all there is to it that's fine, but the hostility for asking is completely unnecessary.

**Comment by gzhou01** [comment]
@thatrandomguyo1 Your comments have not shown anything but complaints without any appreciation for the developers. If you can't see that, then you're just full of yourself.

**Comment by thatrandomguyo1** [comment]
There, comment free since questions aren't allowed.

**Comment by HardcoreHostile** [comment]
I have an iOS device and Windows 10 laptop that I use as the proxy server, if you need help debugging let me know and I'll try and help, I have no coding background do you'll have to walk me through steps of its needed.

**Comment by skroll** [comment]
Well I implemented the capturing of the data and decryption in Go using the proxy. Won't post it here since I know how @kakaroto feels about the decryption keys being in the wild. I'll see if I can use a different lib in the SWProxy code.

**Comment by lstern** [comment]
@skroll current proxy seems to support 'connect' .Everything points to a bug in the proxy code probably related to 'connect' but without being able to use debug the code, I can't really fix it even using external help.

**Comment by Knifetanner2** [comment]
I have tried many times and spent alot of time trying to get the file of my monsters and runes(10+ hrs, probably should've just given up...). I have had no success so  I was wondering if one of the mods who has an android phone could get the file for me. I would give hive information to my account.

**Comment by kakaroto** [comment]
@Knifetanner2 don't give your information to anyone, it's simply not worth it.

@thatrandomguyo1 I agree with you that saying "shut your mouth" is not an acceptable way to talk to others, and no one is telling you not to ask questions or to post here, however your comment was viewed (clearly by more than one person) as arrogant and self-entitled and helped no one. Saying "I see that is bug is still open, is there anything I can do to help you in fixing it?" or even "is there a reason why this bug hasn't been fixed yet?" is still better than trying to use sarcasm to let us know that your urgent need for the fix is more important than global warming.

@skroll @lstern yeah, CONNECT method is supported, and I know it used to work, maybe it stopped working recently? I have no idea. Using a proper proxy might be a good idea, I actually just googled "python proxy server" and copied the first implementation I found, it was a single file that did the job for the proof of concept I was looking to do. I'm sure there are better alternatives out there. If you want to port it to a different or more standard/complete proxy library, feel free to do it.
Honestly, it was FRR day yesterday and I didn't even use the proxy/optimizer, I think the last time I used SWProxy was a couple of months ago, and I'm swamped with work so I spend most of my free time away from the PC, so you understand my motivation is a little low on this right now. But feel free to fix it any way you see fits.
I'd vote for : try to fix the current proxy.py, if it seems overly complicated or takes more than ~30 minutes for a simple fix, then move for a port to a different library (or do both, if better library is available, use it instead, otherwise, default back to the local/broken proxy.py)

**Comment by KooGitKoo** [comment]
I have mac/ios and have had the same problems, but i just tried the work around from @knifesk and it worked like a charm. Thank you for you all amazing people here!

> @daddyrubin > @knifesk said > If you double tap the home screen as soon as the "Connecting to game server" message appears and then switch to settings and set the proxy up, everything works as expected

I doubled tap home screen as soon as I saw "Downloading necessary information...." and went to set the proxy then got back in.

**Comment by aureus1** [comment]
I've done some poking around - It looks like the issue is that the device/registered request never actually gets marked as complete.

I added some logic to mark it as complete (well, right now -any- post with a 0 content-length) and the parser is able to successfully grab my runes/monsters on ios, but it's screwing up the actual game.  Not familiar enough with the HTTP protocol to know -why- it's not being marked as complete, going to have to screw around in wireshark and do some googling to dig further into it.

**Comment by lstern** [comment]
Could you share the code ? Maybe we could do a dirt fix to mark only this
request then we can find the actual bug

**Comment by aureus1** [comment]
Hack fix pull requested; I'll take a look at something less...horrible...after HoH is over.  gotta get back to farming :)

**Comment by lstern** [comment]
Thanks @aureus1 ! I  Did some tests with android and this hack isn't breaking anything (did you experience any problem with this hack ?). As for the package build, are you using mac, linux  or windows ?

**Comment by aureus1** [comment]
No issues that I've seen; game seems fine once it's loaded.

Usually I'm on mac or windows, depending on which machine I happen to grab.

**Comment by lstern** [comment]
accepted the pull request. Created a custom build [here](https://github.com/lstern/SWProxy-plugins/releases) (windows-only). I'll create an official relase once someone confirm that this is working and someone creates a MacOS build.

**Comment by lstern** [comment]
@aureus1 I'm building this using pyinstaller (pyinstaller SWProxy.spec)

**Comment by aureus1** [comment]
OSX build here: https://github.com/aureus1/SWProxy/releases/download/untagged-0c9e8b89ff3155e1ac35/SWProxy

**Comment by lstern** [comment]
404

**Comment by aureus1** [comment]
Try this one, looks like i forgot to publish it: https://github.com/aureus1/SWProxy/releases/download/ios_Fix/SWProxy

**Comment by lstern** [comment]
hum, file is very small.seems like its not including Qt. Previous release
had 14 MB

**Comment by aureus1** [comment]
Downloading + installing pyqt4 now; going to take a while over my cell phone connection (like it might be overnight - not available as a pip package, several 100MB worth of binary downloads).  I'll post the binary when it finishes unless someone else gets to it first

**Comment by Nacheen81** [comment]
I manually added the lines of code for the fix created by @aureus1 to the .99.1 release running in a Linux environment and it works with no errors for iOS and Android devices.  Hope that is helpful in any way.  

Thanks for the work on the fix!

**Comment by aureus1** [comment]
Got PyQt4 set up; new binary here: https://github.com/aureus1/SWProxy/releases/download/ios_Fix/SWProxy

15MB, looks like it has QT pulled in.

**Comment by dotneB** [comment]
Confirmed! Tested master 7f0d02f (Merge pull request #89 from aureus1/fix_ios_issue) with my iOS device and was able to launch the game 5 times out of 5

**Comment by kupoback** [comment]
I downloaded this PyQt4 and the file that @aureus1 posted in his last comment, but don't know what I'm doing... Any suggestions? iOS, and Windows 10 here

**Comment by lstern** [comment]
Released a new version. @kupoback aureus1 file is a MacOS file.

**Comment by kupoback** [comment]
Ah okay. I have a mac as well. However, still unsure what to do. I'll just wait as things wrap up for a .exe or .dmg

**Comment by Xzandro** [comment]
There already is a .exe: https://github.com/kakaroto/SWProxy/releases/tag/0.99.2

**Comment by kupoback** [comment]
Ah, that I didn't see the link to. I just saw 0.99.1 Thank you. Once the game comes back online (can't seem to connect by default), I'll give this a try. Thank you

**Comment by kupoback** [comment]
Cannot connect on either Mac or PC. Fails instantly.

**Comment by HardcoreHostile** [comment]
A big thank you to everyone who has worked on this, it is now working for me! 

Windows 10 marked it as an unknown program and didn't before, but clicked the "More info" and it allowed me to run it, then had to make sure the Firewall exceptions were in place.

Thank you all!
